stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+    }
+
+    // [time.clock.utc]
+
+    class utc_clock;
+    template <class _Duration>
+    using utc_time    = time_point<utc_clock, _Duration>;
+    using utc_seconds = utc_time<seconds>;
+
+    struct leap_second_info {
+        bool is_leap_second;
+        seconds elapsed;
+    };
+
+    template <class _Duration>
+    _NODISCARD leap_second_info get_leap_second_info(const utc_time<_Duration>& _Time) {
+        const utc_seconds _Time_floor = _CHRONO floor<seconds>(_Time);
+        const auto& _Tzdb             = _CHRONO get_tzdb();
+        const auto& _Ls_vector        = _Tzdb.leap_seconds;
+
+        // Find first leap second after _Time and the accumulated offset of all *prior* leap seconds.
+        seconds _Offset;
+        vector<leap_second>::const_iterator _It;
+        if (_Tzdb._All_ls_positive) {
+            // _It here is actually the *2nd* leap second after _Time for the _Offset-1 seconds before the next leap
+            // second insertion. This doesn't overlap with the actual leap second insertion period, so prev(_It) can
+            // always be used to detect if _Time is during an insertion. _Offset, however, needs to be corrected later.
+            // For example, if leap seconds are {100, 200, 300, 400}, we have
+            //
+            //  UTC   sys   *_It   _Offset
+            //   99    99    100         0
+            //  100     X    200         1
+            //  101   100    200         1
+            //  102   101    200         1
+            //  ...
+            //  199   198    200         1
+            //  200   199    300         2^
+            //  201     X    300         2
+            //  202   200    300         2
+            //  ...
+            //  299   297    300         2
+            //  300   298    400         3^
+            //  301   299    400         3^
+            //  302     X    400         3
+            //  303   300    400         3
+            //
+            //  ^_It points to 2nd leap second and _Offset value is wrong.
+
+            _It = _STD upper_bound(_Ls_vector.begin(), _Ls_vector.end(), sys_seconds{_Time_floor.time_since_epoch()});
+            _Offset = seconds{_It - _Ls_vector.begin()};
+        } else {
+            _Offset = seconds{0};
+            for (_It = _Ls_vector.begin(); _It != _Ls_vector.end(); ++_It) {
+                const auto _New_offset = _Offset + _It->value();
+                // Positive leap seconds are counted one second earlier, when insertion begins.
+                if (utc_seconds{_It->date().time_since_epoch()} + (_It->_Positive() ? _Offset : _New_offset)
+                    > _Time_floor) {
+                    break;
+                }
+                _Offset = _New_offset;
+            }
+        }
+
+        if (_It != _Ls_vector.begin()) {
+            // Convert to the last leap second before or equal to _Time (except for the special case described above).
+            const auto _Last_leap       = &*(--_It);

```

Comment:
Parentheses here are unnecessary (there is no visual ambiguity as there is no other way this could be parsed).